cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowStatsSLAHttpHandler.java

```diff
@@ -89,4 +108,131 @@ public void workflowStats(HttpRequest request, HttpResponder responder,
     }
     responder.sendJson(HttpResponseStatus.OK, workflowStatistics);
   }
+
+  @GET
+  @Path("apps/{app-id}/workflows/{workflow-id}/runs/{run-id}/statistics")
+  public void workflowRunDetail(HttpRequest request, HttpResponder responder,
+                                @PathParam("namespace-id") String namespaceId,
+                                @PathParam("app-id") String appId,
+                                @PathParam("workflow-id") String workflowId,
+                                @PathParam("run-id") String runId,
+                                @QueryParam("count") int count,
+                                @QueryParam("interval") String interval) throws Exception {
+    long timeInterval = TimeMathParser.resolutionInSeconds(interval);
+    Id.Workflow workflow = Id.Workflow.from(Id.Namespace.from(namespaceId), appId, workflowId);
+    Collection<WorkflowDataset.WorkflowRunRecord> workflowRunRecords =
+      store.retrieveSpacedRecords(workflow, runId, count, timeInterval).values();
```

Comment:
well, here you get a collection, but the signature in ``DefaultStore`` its Map..